common/src/main/java/io/pravega/common/util/AsyncIterator.java

```diff
@@ -76,4 +78,25 @@
             return result;
         };
     }
+
+    /**
+     * Processes the remaining elements in the AsyncIterator until the specified {@link Predicate} returns false.
+     *
+     * @param until A Predicate that will be used to the decide the number of elements on which the Consumer will be invoked.
+     * @param consumer A Consumer that will be invoked for each remaining element. The consumer will be invoked by the same thread, hence
+     *                any new invocation will will wait until the previous invocation is complete.
+     *                 given Executor, but any new invocation will wait for the previous invocation to complete.
+     * @return A CompletableFuture that, when completed, will indicate that the processing is complete.
+     */
+    default CompletableFuture<Void> collectUntil(Consumer<? super T> consumer, Predicate<? super T> until) {
```

Comment:
You can merge both `consumer` and `until` into one . There is absolutely no need to have both.

Reply:
done.